### PR TITLE
update signers and add sender for rpc api

### PR DESCRIPTION
**Pull request by ProDog**
- add signers for send asset api
- supports address and script hash
- add sender for invoke contract api

### CODE REVIEW

File: src/RpcServer/RpcServer.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <Version>3.0.0-preview3</Version>
+    <Version>3.0.0-preview2</Version>
```
Comment (shargon) [inline src/RpcServer/RpcServer.csproj:4]:
```suggestion
    <Version>3.0.0-preview3</Version>
```

**Comment by ProDog** [comment]
Contract calls support custom scope, like `invokefunction` and `invokescript`.
For the wallet transfer function, I think `CalledByEntry` is enough, it is easier for users, and they can call it by `invoke` api if they want to customize it.

File: src/RpcServer/RpcServer.Wallet.cs [hunk 2]
```diff
@@ -299,5 +311,21 @@ private JObject SignAndRelay(Transaction tx)
                 return context.ToJson();
             }
         }
+
+        internal static UInt160 TryScriptHash(string input)
```
Comment (Tommo-L) [inline src/RpcServer/RpcServer.Wallet.cs:315]: Need a better name, `TryXXXX` should return bool value.

File: src/RpcServer/RpcServer.Wallet.cs [hunk 3]
```diff
@@ -299,5 +311,21 @@ private JObject SignAndRelay(Transaction tx)
                 return context.ToJson();
             }
         }
+
+        internal static UInt160 StringToScriptHash(string input)
```
Comment (vncoelho) [inline src/RpcServer/RpcServer.Wallet.cs:315]: Maybe it is better to create a function just for AssetID and other function for scripthash.

Comment (ProDog) [reply src/RpcServer/RpcServer.Wallet.cs]: To AssetID can use UInt160.Parse() directly, now this just for scripthash.

**Comment by ProDog** [comment]
Can merge?

**Review by shargon — APPROVED** [review]

**Comment by shargon** [comment]
The rpc specifications will need to be updated.

**Review by Tommo-L — APPROVED** [review]

**Comment by Tommo-L** [comment]
> The rpc specifications will need to be updated.

I have opened the issue https://github.com/neo-project/docs/issues/1129, we'll update it later.

**Pull request merged**